### ISSUE TITLE
Enhancement: Add rule docs recommendation metadata for IDE sorting

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#38]
## Description

ESLint supports `meta.docs.recommended` property to indicate whether a rule should be included in recommended configs. Adding this (and other meta properties) would improve IDE integration and rule documentation.

## Suggested Meta Properties

```ts
meta: {
  docs: {
    description: '...',
    recommended: 'strict' | 'error' | 'warn' | false,
    url: 'https://github.com/pertrai1/eslint-plugin-llm-core/blob/main/docs/rules/xxx.md',
  },
  hasSuggestions: boolean,
  fixable: 'code' | 'whitespace' | null,
  type: 'problem' | 'suggestion' | 'layout',
}
```

## Benefits

- IDEs can sort/prioritize rules
- Documentation generators can categorize rules
- Users can see at a glance which rules are `recommended`
- Better integration with ESLint tooling ecosystem
